pkg/inject/sidecar_test.go

```diff
@@ -231,7 +231,7 @@ func TestSidecarNeeded(t *testing.T) {
 		{
 			dep:    depWithAgent,
 			ns:     ns(map[string]string{}),
-			needed: false,
+			needed: true,
```

Comment:
Why did this change? The `Needed` should return whether an agent is needed in the deployment. The behavior so far has been that this would return `false` if the deployment had an agent already. I think this brings a new question: should we touch sidecars that were added explicitly by the deployment? So far, this hasn't been a problem, as we wouldn't touch sidecars at all after they had been injected.

Perhaps we should change the container name from `jaeger-agent` to `operator-internal-jaeger-agent` (based on https://github.com/open-telemetry/opentelemetry-operator/pull/17#discussion_r389573208)?

Reply:
I modified the Needed to return true even if the agent is already injected because the flags could be changed. With this change I will try to inject the sidecar every time.

Reply:
@jpkrohling  After thought more about this.

Is the name change (or checking the label) necessary? or Can I assume that if the deployment has the annotation is entirely managed by the operator?

Manual or explicit added sidecars won't have an annotation,  are they?

Reply:
Ok, I retract of what I said about the label. It is necessary to distinguish because the namespace injection. So I updated this PR to use the label in the case that we already have a sidecar agent.

Reply:
About tests

I included some extra tests for needed when Test that explicit sidecars are skiped.
Also added the assertion.

Do we need to require other tests? I think for "ensuring that sidecars are actually being updated" there already tests in place.